### PR TITLE
add localization strings for module render-biblio

### PR DESCRIPTION
**Pull request by ThobyV**
An Implementation of the new method for modules localization from this issue https://github.com/w3c/respec/issues/2084 . Big Thanks to @marcoscaceres for the guidelines. I hope I am close to the solution.

### CODE REVIEW

File: src/core/render-biblio.js [hunk 1]
```diff
@@ -4,9 +4,33 @@
 import { addId } from "./utils";
 import hyperHTML from "hyperhtml";
 import { pub } from "./pubsubhub";
+import { lang as defaultLang } from "../core/l10n"
 
 export const name = "core/render-biblio";
 
+const localization_strings = {
```
Comment (marcoscaceres) [inline src/core/render-biblio.js]: We can just call this.. 
```suggestion
const l10n = {
```

File: src/core/render-biblio.js [hunk 2]
```diff
@@ -4,9 +4,33 @@
...
+    norm_references: "Referencias normativas",
+    references: "Referencias"
+  }
+}
+
+const lang = defaultLang in meta ? defaultLang : "en";
+
+const l10n = localization_strings[lang];
+
+
```
Comment (marcoscaceres) [inline src/core/render-biblio.js]: Probably don't need this extra line here. 
```suggestion
```

File: src/core/render-biblio.js (hunk 1 above)
Comment (marcoscaceres) [inline src/core/render-biblio.js]: We tend not to use "_" in variable names... stick to camel case. 
 
```suggestion
const localizationStrings = {
```

File: src/core/render-biblio.js [hunk 3]
```diff
@@ -4,9 +4,33 @@
 import { addId } from "./utils";
 import hyperHTML from "hyperhtml";
 import { pub } from "./pubsubhub";
+import { lang as defaultLang } from "../core/l10n"
```
Comment (marcoscaceres) [inline src/core/render-biblio.js]:
```suggestion
import { lang as defaultLang } from "../core/l10n";
```

File: src/core/render-biblio.js [hunk 4]
```diff
@@ -4,9 +4,33 @@
...
+    references: "Referenties"
+  },
+  es: {
+    info_references: "Referencias informativas",
+    norm_references: "Referencias normativas",
+    references: "Referencias"
+  }
+}
+
+const lang = defaultLang in meta ? defaultLang : "en";
```
Comment (marcoscaceres) [inline src/core/render-biblio.js]: I think here you want: 

```suggestion
const lang = defaultLang in localizationStrings ? defaultLang : "en";
```

Comment (ThobyV) [reply src/core/render-biblio.js]: Oops! I totally skipped. Will change this now

**Comment by marcoscaceres** [comment]
Please check also for any additional errors that TravisCI might be reporting.

**Comment by ThobyV** [comment]
I am sorry for creating a pull without  taking everything into consideration, I will fix all now. Thanks for reviewing @marcoscaceres

**Comment by marcoscaceres** [comment]
We should probably add a test for this too. You can copy:

https://github.com/w3c/respec/blob/develop/tests/spec/core/l10n-spec.js#L30-L41

And and set the language to `nl`... then you can check that the references section of the generated spec has the expected localized text. 

You should add the test to the bottom of:
https://github.com/w3c/respec/blob/develop/tests/spec/core/biblio-spec.js

**Comment by marcoscaceres** [comment]
Please be sure to use https://prettier.io over your code. ES Lint will get upset otherwise. 

You can also do: `eslint --fix src/core/render-biblio.js` to fix issues.

**Comment by ThobyV** [comment]
@marcoscaceres  I do not  have extensive experiences with writing tests. This is what is came up with so far

```javascript
it("should make sure references section has expected localization text", async () => {
  const body = makeDefaultBody();

  const ops = {
    config: makeBasicConfig(),
    htmlAttrs: {
      lang: "nl",
    },
    body,
  };
  const ref = doc.getElementById("references");
  const doc = await makeRSDoc(ops);
  expect(doc.documentElement.lang).toEqual("nl");
  expect(ref.querySelector("h1").textContent).toEqual("Referenties");
});
```
For some reason I suspect is from my test code this test Is failing.

**Comment by marcoscaceres** [comment]
Load up the example/basic.html spec in the browser and modify the `<html>` to be `<html lang="nl">`. That should localize the spec and run your code.  Then take a look at what heading level the `Referenties` is and how to get query it.

**Comment by ThobyV** [comment]
Oops! Never mind. I have found  the issue was that `makeDefaultBody()` only makes a  very basic spec that  does not somehow include references section.

**Comment by ThobyV** [comment]
@marcoscaceres  I have made a new commit. Thank  you so much for reviewing and helping all the while, I hope to get my first merge. :smile: From here I think it will more quicker to update the rest of  the modules and try other first issues.

File: tests/spec/core/biblio-spec.js [hunk 5]
```diff
@@ -209,3 +209,27 @@ describe("W3C — Bibliographic References", () => {
...
+
+it("should make sure references section has expected localization text", async () => {
+  const ops = {
+    config: makeBasicConfig(),
+    htmlAttrs: {
+      lang: "nl",
+    },
+    body: `
+  <section class="informative" id="intro">
+    <h2>
```
Comment (marcoscaceres) [inline tests/spec/core/biblio-spec.js]: See if you can reduce this down to the absolute minimum of what is being tested (i.e., only section tag with the [dom] reference).

File: tests/spec/core/biblio-spec.js [hunk 6]
```diff
@@ -209,3 +209,27 @@ describe("W3C — Bibliographic References", () => {
...
+      This section has a CSS class "<code>informative</code>", so it is
+      listed in the TOC. ReSpec will list this reference as non-normative
+      since this is an informative section [[DOM]].
+    </p>
+  </section>`,
+  };
+  const doc = await makeRSDoc(ops);
+  const ref = doc.getElementById("references");
+  expect(doc.documentElement.lang).toEqual("nl");
+  expect(ref.querySelector("h2").textContent).toMatch("Referenties");
```
Comment (marcoscaceres) [inline tests/spec/core/biblio-spec.js]: Check one of the h3s too, then I think we are good👍

File: src/core/render-biblio.js (hunk 2 above)
Comment (marcoscaceres) [reply src/core/render-biblio.js]: Don’t forget this small nit.

**Comment by ThobyV** [comment]
@marcoscaceres I've been able to make requested changes in a newer commit. :smile: Would be glad to solve any more issues from this P.R.

File: tests/spec/core/biblio-spec.js [hunk 7]
```diff
@@ -209,3 +209,37 @@ describe("W3C — Bibliographic References", () => {
...
+  const ops = {
+    config: makeBasicConfig(),
+    htmlAttrs: {
+      lang: "nl",
+    },
+    body: `<section class="informative" id="intro">[[DOM]]</section>`,
+  };
+  const doc = await makeRSDoc(ops);
+  const ref = doc.getElementById("references");
+  expect(doc.documentElement.lang).toEqual("nl");
```
Comment (marcoscaceres) [inline tests/spec/core/biblio-spec.js]:
```suggestion
  expect(doc.documentElement.lang).toBe("nl");
```

File: tests/spec/core/biblio-spec.js [hunk 8]
```diff
@@ -209,3 +209,37 @@ describe("W3C — Bibliographic References", () => {
...
+it("makes sure the Informative references section has expected localization text", async () => {
+  const ops = {
+    config: makeBasicConfig(),
+    htmlAttrs: {
+      lang: "nl",
+    },
+    body: `<section class="informative" id="intro">[[DOM]]</section>`,
+  };
+  const doc = await makeRSDoc(ops);
+  const ref = doc.getElementById("references");
```
Comment (marcoscaceres) [inline tests/spec/core/biblio-spec.js]: Here, you don't need ref, just grab the h2's textContent directly. 

```suggestion
  const { textContent } = doc.getElementById("#references h2");
```

File: tests/spec/core/biblio-spec.js [hunk 9]
```diff
@@ -209,3 +209,37 @@ describe("W3C — Bibliographic References", () => {
...
+    htmlAttrs: {
+      lang: "nl",
+    },
+    body: `<section>[[DOM]]</section>`,
+  };
+  const doc = await makeRSDoc(ops);
+  const ref = doc.getElementById("references");
+  expect(doc.documentElement.lang).toEqual("nl");
+  expect(ref.querySelector("h2").textContent).toMatch("Referenties");
+  expect(ref.querySelector("section").querySelector("h3").textContent).toMatch(
```
Comment (marcoscaceres) [inline tests/spec/core/biblio-spec.js]: Here we can be a bit more concise:
 
```suggestion
  const {textContent: h3TextContent} = doc.querySelector("#references h3");
  expect(h3TextContent.trim()).toBe("Normatieve referenties");
```

File: tests/spec/core/biblio-spec.js [hunk 10]
```diff
@@ -209,3 +209,37 @@ describe("W3C — Bibliographic References", () => {
...
+    config: makeBasicConfig(),
+    htmlAttrs: {
+      lang: "nl",
+    },
+    body: `<section class="informative" id="intro">[[DOM]]</section>`,
+  };
+  const doc = await makeRSDoc(ops);
+  const ref = doc.getElementById("references");
+  expect(doc.documentElement.lang).toEqual("nl");
+  expect(ref.querySelector("h2").textContent).toMatch("Referenties");
```
Comment (marcoscaceres) [inline tests/spec/core/biblio-spec.js]:
```suggestion
  expect(textContent.trim()).toBe("Referenties");
```

File: tests/spec/core/biblio-spec.js (hunk 8 above)
Comment (ThobyV) [reply tests/spec/core/biblio-spec.js]: I tried this too...seems not to work with the test spec I applied to `body`. But I will try to figure out why and apply your suggestions now. Thanks.

Comment (marcoscaceres) [reply tests/spec/core/biblio-spec.js]: Sorry, should be: 
const { textContent } = doc.querySelector("#references h2");

File: tests/spec/core/biblio-spec.js (hunk 10 above)
Comment (ThobyV) [reply tests/spec/core/biblio-spec.js]: @marcoscaceres I've made requested changes. However I used `substr()` to remove space and unwanted characters from `textContent` as `trim()` only removed white space and caused test to fail.

**Comment by marcoscaceres** [comment]
Seems your branch has fallen out of sync with the main (develop) branch? You might need to rebase or apply your changes to a new branch.

File: tests/spec/core/biblio-spec.js [hunk 11]
```diff
@@ -212,3 +209,23 @@ describe("W3C — Bibliographic References", () => {
...
+    body: `
+    <section class="informative" id="intro">[[DOM]]</section>
+    <section>[[HTML]]</section>
+    `,
+  };
+  const doc = await makeRSDoc(ops);
+  const { textContent } = doc.querySelector("#references h2");
+  const [normRef, infoRef] = doc.querySelectorAll("#references h3");
+  expect(doc.documentElement.lang).toBe("nl");
+  expect(textContent.substr(3)).toBe("Referenties");
```
Comment (marcoscaceres) [inline tests/spec/core/biblio-spec.js]: oh, forgot about this... just test whatever the string is without doing "substr".

Comment (ThobyV) [reply tests/spec/core/biblio-spec.js]: would It be ok if I tested for a match like  `expect(textContent).toMatch("Referenties")` as this works actually.

File: tests/spec/core/biblio-spec.js [hunk 12]
```diff
@@ -212,3 +209,23 @@ describe("W3C — Bibliographic References", () => {
...
+    <section class="informative" id="intro">[[DOM]]</section>
+    <section>[[HTML]]</section>
+    `,
+  };
+  const doc = await makeRSDoc(ops);
+  const { textContent } = doc.querySelector("#references h2");
+  const [normRef, infoRef] = doc.querySelectorAll("#references h3");
+  expect(doc.documentElement.lang).toBe("nl");
+  expect(textContent).toMatch("Referenties");
+  expect(normRef.textContent).toMatch("Normatieve referenties");
```
Comment (marcoscaceres) [inline tests/spec/core/biblio-spec.js]: Use `toContain()` instead, as toMatch() is for regular expressions.

Comment (ThobyV) [reply tests/spec/core/biblio-spec.js]: Oh ok @marcoscaceres. got  it! Would I need to do a squash for my commits?

**Comment by marcoscaceres** [comment]
You’ll need to rebase off develop somehow, but I’m not sure about the state of your branch (it doesn’t look good 😬). If you are new to git, I would just copy the files you change out and make a new branch. Then paste the changed files into that branch. The send the new branch as a new pull request.

**Comment by ThobyV** [comment]
I was thinking so too :d I made a mistake while trying to rebase. I guess I'll get better with time. So i guess to close this issue?

**Pull request closed without merging**